packages/untp-test-suite/src/core/services/dynamic-loading-schemas/loadingSchema.service.ts

```diff
@@ -0,0 +1,26 @@
+import { IDynamicLoadingSchemaService } from './types';
+import { checkSchemaExists, checkSchemaVersionExists, getSchemaContent } from './utils';
+
+export const dynamicLoadingSchemaService: IDynamicLoadingSchemaService = async (schema, version) => {
+  try {
+    const isValidSchema = await checkSchemaExists(schema);
+    if (!isValidSchema) {
+      throw new Error(`Schema not found`);
+    }
+
+    const isValidVersion = await checkSchemaVersionExists(schema, version);
+    if (!isValidVersion) {
+      throw new Error(`Version not found for schema ${schema}`);
+    }
+
+    const content = await getSchemaContent(schema, version);
+    if (!content) {
+      throw new Error(`Content in ${schema} schema not found`);
+    }
+
+    return JSON.parse(content);
+  } catch (e) {
+    const error = e as Error;
+    throw new Error(error.message ?? 'Error loading schema');
+  }
```

Comment:
I don't think we should use a try-catch block here because the error is thrown in the try block, which will then be caught in the catch block. As you implemented, you throw it again in the catch block. Logically, it's no different.